src/sphinxcontrib/towncrier/__init__.py

```diff
@@ -53,18 +53,20 @@ def _get_changelog_draft_entries(
         config_path: str = None,
 ) -> str:
     """Retrieve the unreleased changelog entries from Towncrier."""
+    if target_version.startswith('|') and target_version.count('|') > 1:
```

Comment:
@davfsa I suggest moving this escaping into a separate pure function so that we could write a test for it specifically. This bit is obscure and easy enough to break unintentionally so I'm now convinced that for this PR to be truly valuable, there must be a regression test bundled with it. Sounds good?

Reply:
Sounds great, will do when I get around to! Hopefully later today or early tomorrow

Reply:
Thanks! No rush tho :)

Reply:
Done!